events/admin.py

```diff
@@ -24,6 +24,8 @@ class AutoIdBaseAdmin(BaseAdmin):
     def save_model(self, request, obj, form, change):
         system_id = settings.SYSTEM_DATA_SOURCE_ID
         obj.data_source_id = system_id
+        if obj.origin_id:
+            obj.id = ':'.join([system_id, obj.origin_id])
```

Comment:
One more thing :D Here, what might happen (an edge case, I admit) is that we are saving an existing object. In that case, if id exists, we definitely should not change the id. Therefore, construct a new id *only* when id is missing, i.e. move these two lines inside the `if not obj.id` clause.

Reply:
Hm, actually best move *all* the lines inside the clause. We don't want to touch origin_id either if the object has been created earlier.

Reply:
Done.